File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/model/Alert.java
package com.lattis.ellipse.domain.model;

public enum Alert {

    OFF,THEFT,CRASH;

    private String lockId;

    public void setLockId(String lockId) {
        this.lockId = lockId;
    }

    public String getLockId() {
        return lockId;
    }

    public static Alert forValue(String value){
        for(Alert alert:values()){
            if(alert.name().equals(value)){
                return alert;
            }
        }
        return null;
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/models/Alert.kt
package com.lattis.domain.models

enum class Alert {

    OFF, THEFT, CRASH;

    var lockId: String? = null

    companion object {

        fun forValue(value: String): Alert? {
            for (alert in values()) {
                if (alert.name == value) {
                    return alert
                }
            }
            return null
        }
    }
}
